### PR TITLE
Prefer native MacOS code over SDL on OSX

### PR DESCRIPTION
**Pull request by morguldir**
This will make opentk prefer running with native code when running under macos

---

Fix crashes on macOS when having SDL2 installed

### CODE REVIEW

File: osu.Framework/Host.cs [hunk 1]
```diff
@@ -6,17 +6,21 @@
...
 using System;
+using OpenTK;
 
 namespace osu.Framework
 {
     public static class Host
     {
         public static DesktopGameHost GetSuitableHost(string gameName, bool bindIPC = false)
         {
+            var toolkitOptions = new ToolkitOptions { EnableHighResolution = true};
```
Comment (VPeruS) [inline osu.Framework/Host.cs]: Space before }

File: osu.Framework/Platform/Linux/LinuxGameHost.cs [hunk 2]
```diff
@@ -2,18 +2,19 @@
...
-        internal LinuxGameHost(string gameName, bool bindIPC = false)
-            : base(gameName, bindIPC)
+        internal LinuxGameHost(string gameName, bool bindIPC = false, ToolkitOptions toolkitOptions = null)
+            : base(gameName, bindIPC, toolkitOptions)
         {
             Window = new LinuxGameWindow();
             Window.WindowStateChanged += (sender, e) =>
             {
-                if (Window.WindowState != OpenTK.WindowState.Minimized)
+                if (Window.WindowState != WindowState.Minimized)
```
Comment (VPeruS) [inline osu.Framework/Platform/Linux/LinuxGameHost.cs]: Why not in separate pr? (1)

File: osu.Framework/Platform/MacOS/MacOSGameHost.cs [hunk 3]
```diff
@@ -4,18 +4,19 @@
...
-        internal MacOSGameHost(string gameName, bool bindIPC = false)
-            : base(gameName, bindIPC)
+        internal MacOSGameHost(string gameName, bool bindIPC = false, ToolkitOptions toolkitOptions = null)
+            : base(gameName, bindIPC, toolkitOptions)
         {
             Window = new MacOSGameWindow();
             Window.WindowStateChanged += (sender, e) =>
             {
-                if (Window.WindowState != OpenTK.WindowState.Minimized)
+                if (Window.WindowState != WindowState.Minimized)
```
Comment (VPeruS) [inline osu.Framework/Platform/MacOS/MacOSGameHost.cs]: ditto 1

File: osu.Framework/Platform/Windows/WindowsGameHost.cs [hunk 4]
```diff
@@ -30,7 +31,7 @@ internal WindowsGameHost(string gameName, bool bindIPC = false)
 
         private void onWindowOnWindowStateChanged(object sender, EventArgs e)
         {
-            if (Window.WindowState != OpenTK.WindowState.Minimized)
+            if (Window.WindowState != WindowState.Minimized)
```
Comment (VPeruS) [inline osu.Framework/Platform/Windows/WindowsGameHost.cs]: ditto 1

File: osu.Framework/Platform/Windows/WindowsGameHost.cs [hunk 5]
```diff
@@ -16,8 +17,8 @@ public class WindowsGameHost : DesktopGameHost
 
         public override bool CapsLockEnabled => Console.CapsLock;
 
-        internal WindowsGameHost(string gameName, bool bindIPC = false)
-            : base(gameName, bindIPC)
+        internal WindowsGameHost(string gameName, bool bindIPC = false, ToolkitOptions toolkitOptions = null)
```
Comment (VPeruS) [inline osu.Framework/Platform/Windows/WindowsGameHost.cs]: ditto 2

File: osu.Framework/Platform/DesktopGameHost.cs [hunk 6]
```diff
@@ -20,8 +21,8 @@ public abstract class DesktopGameHost : GameHost
         private readonly TcpIpcProvider ipcProvider;
         private readonly Thread ipcThread;
 
-        protected DesktopGameHost(string gameName = @"", bool bindIPCPort = false)
-            : base(gameName)
+        protected DesktopGameHost(string gameName = @"", bool bindIPCPort = false, ToolkitOptions toolkitOptions = null)
```
Comment (VPeruS) [inline osu.Framework/Platform/DesktopGameHost.cs]: ToolkitOptions toolkitOptions = default(ToolkitOptions) (2)

File: osu.Framework/Platform/GameHost.cs [hunk 7]
```diff
@@ -159,9 +159,9 @@ public double MaximumInactiveHz
 
         public DependencyContainer Dependencies { get; } = new DependencyContainer();
 
-        protected GameHost(string gameName = @"")
+        protected GameHost(string gameName = @"", ToolkitOptions toolkitOptions = null)
```
Comment (VPeruS) [inline osu.Framework/Platform/GameHost.cs]: ditto 2

File: osu.Framework/Platform/Linux/LinuxGameHost.cs [hunk 8]
```diff
@@ -2,18 +2,19 @@
...
 using osu.Framework.Platform.Linux.Native;
+using OpenTK;
 
 namespace osu.Framework.Platform.Linux
 {
     public class LinuxGameHost : DesktopGameHost
     {
-        internal LinuxGameHost(string gameName, bool bindIPC = false)
-            : base(gameName, bindIPC)
+        internal LinuxGameHost(string gameName, bool bindIPC = false, ToolkitOptions toolkitOptions = null)
```
Comment (VPeruS) [inline osu.Framework/Platform/Linux/LinuxGameHost.cs]: ditto 2

File: osu.Framework/Platform/Linux/LinuxGameHost.cs (hunk 2 above)
Comment (VPeruS) [reply osu.Framework/Platform/Linux/LinuxGameHost.cs]: Yeah but it's not - "Prefer native MacOS code over SDL on OSX"

Comment (peppy) [reply osu.Framework/Platform/Linux/LinuxGameHost.cs]: This is fine as-is. We allow such code-refactors in PRs. If we didn't this world would fall apart.

**Comment by smoogipoo** [comment]
Please check the messages which appveyor gives (https://ci.appveyor.com/project/peppy/osu-framework/builds/19623431/messages), just remove the type specification from the `default`.

We should change these to be warnings.

**Comment by peppy** [comment]
I wonder if this should also be applied to windows?

**Pull request merged**